From 318b932d5b9cfc1ffcc21d4594fd41f834f993d3 Mon Sep 17 00:00:00 2001
From: Basanagouda Nagappa Koppad <Basanagouda.Nagappa.Koppad@intel.com>
Date: Wed, 20 Aug 2025 05:56:21 +0000
Subject: [PATCH] [TWL] Shift storageproxyd to a very early stage

Gatekeeper or Keymaster may call SecureStorage before /data
are fully decrypted or mounted. In some conditions, it may
cause to deadlock. Shift storageproxyd booting as earlier
as possible can break the deadlock.

Tracked-On: OAM-133804
Signed-off-by: Basanagouda Nagappa Koppad <Basanagouda.Nagappa.Koppad@intel.com>

diff --git a/groups/tee/trusty/init.rc b/groups/tee/trusty/init.rc
index 19739ced..fcac96e7 100644
--- a/groups/tee/trusty/init.rc
+++ b/groups/tee/trusty/init.rc
@@ -1,6 +1,12 @@
 on post-fs-data
     mkdir /data/vendor/securestorage 0700 system system
 
+on early-fs
+    start storageproxyd
+
+service storageproxyd /vendor/bin/storageproxyd -d /dev/trusty-ipc-dev0 -p /data/vendor/securestorage -r /dev/vport0p1 -t virt    user system
+    group system
+
 on boot
     start keyboxd
 
-- 
2.34.1

